src/sentry/spans/consumers/process/flusher.py

```diff
@@ -26,6 +27,74 @@
 logger = logging.getLogger(__name__)
 
 
+class MultiProducerManager:
+    """
+    Manages multiple Kafka producers for load balancing across brokers/topics.
+
+    Configure multiple producers in settings.py:
+
+    KAFKA_MULTI_PRODUCER_CONFIGS = {
+        "buffered-segments": [
+            {"cluster": "default", "topic": "buffered-segments-1"},
+            {"cluster": "secondary", "topic": "buffered-segments-2"}
+        ]
+    }
+    """
+
+    def __init__(self, topic: Topic):
+        self.topic = topic
+        self.producers: list[KafkaProducer] = []
+        self.topics: list[ArroyoTopic] = []
+        self.current_index = 0
+        self._setup_producers()
+
+    def _setup_producers(self):
+        """Setup producers based on configuration."""
+        # Get multi-producer config if available, otherwise use single producer
+        multi_config = getattr(settings, "KAFKA_MULTI_PRODUCER_CONFIGS", {}).get(self.topic.value)
+
+        if multi_config:
+            # Multiple producers configured
+            for config in multi_config:
+                cluster_name = config["cluster"]
+                topic_name = config.get(
+                    "topic", get_topic_definition(self.topic)["real_topic_name"]
+                )
```

Comment:
So, the cellularized topics are going to be only physical topics without an equivalent in the schema registry, is this correct? 
If yes, what was the rationale to implement it this way rather than creating explicitly one topic per cell in the schema registry? We never ran into this scenario after we introduced the Kafka control plane so it is important to provide consistent directions for the next time. 
Also how are you planning to manage this in the consumer? Will there be a MultiConsumer that maps the topic name to the physical one differently fopr each cell ?

Reply:
I am ok with this approach. 
What about the consumer ?

Reply:
> Also how are you planning to manage this in the consumer? Will there be a MultiConsumer that maps the topic name to the physical one differently fopr each cell ?

Reply:
we likely will use `--topic` on `sentry run consumer` to entirely override the physical topic for certain deployment.

Reply:
> we likely will use --topic on sentry run consumer to entirely override the physical topic for certain deployment.

Please let's have a config driven override. So we can ensure the config of the producer matches the one for the consumer.  The CLI can take a `cell-id` argument
If we use the `--topic` cli argument we will ahve the consumer topics stated in a different place (different repo). That's going to cause mistakes